src/main/java/com/bbmk/restaurant/Dao/RestaurantDataAccessService.java

```diff
@@ -188,12 +197,71 @@ public void insertFood(Integer id, String name, Integer price) {
 	@Query("DELETE FROM Food WHERE id = :id")
 	public void deleteFoodById(Long id) {
 
+
 	}
 
 	@Override
 	@Query("DELETE FROM Food WHERE id = :id")
 	public void deleteFood(Integer id, String name, Integer price) {
 
+	}
+
+	@Override
+	public void updateFood(Food food) {
+
+	}
+
+	@Override
+	public void updateFoodbyID(Integer id) {
+
+	}
+
+	@Override
+	public void updateFoodbyName(String name) {
+
+	}
+
+	@Override
+	public void updateFoodbyPrice(Integer price) {
+
+	}
+
+	@Override
+	@Query("UPDATE Food SET name = :name, price = :price WHERE id = :id")
+	public void updateFood(Integer id, String name, Integer price) {
+		if (id == null) {
+			throw new IllegalArgumentException("id is null");
+		}else if (name == null) {
+			throw new IllegalArgumentException("name is null");
+		}else if (price == null) {
+			throw new IllegalArgumentException("price is null");
+		}
+
+
+	}
+
+	@Override
+	@Query("UPDATE Food SET price = :price WHERE id = :id")
+	public void updateFoodPrice(Integer id, Integer price) {
```

Comment:
This approach wont work, but well-tried, we want to update the `price` and `name` once with one method